WebCore\DerivedSources\JSReadableStreamSource.cpp: warning: unused function 'jsReadableStreamSource_controller'

iangrunert opened the pull request:
#### e693585f93abd942ea70668feb1ca483234b25b9
<pre>
WebCore\DerivedSources\JSReadableStreamSource.cpp: warning: unused function &apos;jsReadableStreamSource_controller&apos;
<a href="https://bugs.webkit.org/show_bug.cgi?id=260376">https://bugs.webkit.org/show_bug.cgi?id=260376</a>

Reviewed by Fujii Hironori.

Emit UNUSED_PARAM for the attributeGetterName to prevent compiler
warning in clang-cl.

* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(GenerateAttributeGetterTrampolineDefinition):

Canonical link: <a href="https://commits.webkit.org/267420@main">https://commits.webkit.org/267420@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5bd0edcc6d1f062a6dc6361a44046667fd5d3660

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/16493 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/16815 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/17254 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/18272 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/15467 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/20046 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/16958 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/17820 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/16690 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/17107 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/14266 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/19049 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/14351 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/14950 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/21746 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/14230 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/15339 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/15116 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/19430 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/15730 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/15707 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/13339 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/18050 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/14908 "Built successfully") | | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/4246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3958 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/19277 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/19270 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/15533 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/4076 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

iangrunert commented on Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
```diff
@@ -5545,8 +5545,9 @@ sub GenerateAttributeGetterTrampolineDefinition
     }
     
-    push(@$outputArray, "JSC_DEFINE_CUSTOM_GETTER(${attributeGetterName}, (JSGlobalObject* lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName))\n");
+    push(@$outputArray, "static inline JSC_DEFINE_CUSTOM_GETTER(${attributeGetterName}, (JSGlobalObject* lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName))\n");
```
Looks like this change causes the bindings tests to fail, will see if I can get it to build without the `static inline` addition.

fujii commented on Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
```diff
@@ -5545,7 +5545,7 @@ sub GenerateAttributeGetterTrampolineDefinition
     }
     
-    push(@$outputArray, "JSC_DEFINE_CUSTOM_GETTER(${attributeGetterName}, (JSGlobalObject* lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName))\n");
+    push(@$outputArray, "REFERENCED_FROM_ASM JSC_DEFINE_CUSTOM_GETTER(${attributeGetterName}, (JSGlobalObject* lexicalGlobalObject, EncodedJSValue thisValue, PropertyName attributeName))\n");
```
We don't need this change because JSC_DECLARE_CUSTOM_GETTER already contains REFERENCED_FROM_ASM.

iangrunert replied:
Yeah you're right - the code compiles without this change. I must be missing something though because I can't see where `REFERENCED_FROM_ASM` is added for `JSC_DEFINE_CUSTOM_GETTER` (`DEFINE`, not `DECLARE`).

`JSC_DECLARE_CUSTOM_GETTER` contains `REFERENCED_FROM_ASM` via `JSC_DECLARE_JIT_OPERATION_WITHOUT_WTF_INTERNAL`.

`JSC_DEFINE_CUSTOM_GETTER` uses `JSC_DEFINE_JIT_OPERATION_WITHOUT_VARIABLE` which doesn't contain `REFERENCED_FROM_ASM`.

fujii replied:
Right. it's enough if the declaration has the attribute.

fujii commented on Source/WTF/wtf/Compiler.h:
```diff
@@ -393,7 +393,7 @@
 /* REFERENCED_FROM_ASM */
 
-#if !defined(REFERENCED_FROM_ASM) && COMPILER(GCC_COMPATIBLE)
+#if !defined(REFERENCED_FROM_ASM) && (COMPILER(GCC) || COMPILER(CLANG))
```
We need one more change for https://bugs.webkit.org/show_bug.cgi?id=260689 .
We have to consider enabling WTF_COMPILER_GCC_COMPATIBLE .

fujii replied:
I tried enabling WTF_COMPILER_GCC_COMPATIBLE locally.
I learned that it's not very easy. Never mind WTF_COMPILER_GCC_COMPATIBLE at the moment.

iangrunert committed:
WebCore\DerivedSources\JSReadableStreamSource.cpp: warning: unused function 'jsReadableStreamSource_controller'
https://bugs.webkit.org/show_bug.cgi?id=260376

Reviewed by Fujii Hironori.

Emit UNUSED_PARAM for the attributeGetterName to prevent compiler
warning in clang-cl.

* Source/WebCore/bindings/scripts/CodeGeneratorJS.pm:
(GenerateAttributeGetterTrampolineDefinition):

Canonical link: https://commits.webkit.org/267420@main
